crates/walrus-service/src/node/storage/shard.rs

```diff
@@ -137,6 +138,31 @@ impl ShardStatus {
     }
 }
 
+#[derive(Debug)]
+struct ShardStatusState {
+    shard_status: DBMap<(), ShardStatus>,
+}
+
+impl ShardStatusState {
+    pub fn update_status(&mut self, new_status: ShardStatus) -> Result<(), TypedStoreError> {
+        self.shard_status.insert(&(), &new_status)?;
+        Ok(())
+    }
+
+    pub fn get_status(&self) -> Result<Option<ShardStatus>, TypedStoreError> {
+        self.shard_status.get(&())
+    }
+
+    /// Executes a function with mutable access to the shard_status DBMap.
+    /// This allows atomic operations on the shard status within a critical section.
```

Comment:
Question: This assumes that the column family wasn't opened multiple times, correct?

Reply:
Also, question: When we enable transactions on the DB, we could use those instead, right?

Reply:
If you are talking about opening same column family multiple times from RocksDb, I don't think RocksDb currently allows that. However, if you are talking about opening the column family once, but have a reference to the DBMap in multiple places, and not all through the new ShardStatusState, yes, here the code assumes that.

> When we enable transactions on the DB, we could use those instead, right?
Yes! The need here is to do a read-modify-write transaction. If not implemented in the application logic, another way to do this is to use merge operator, which will further complicate the code. Since the logic of the shard status is simple, I just guarded the access to it using a mutex to achieve transaction.

But I agree that this can be replaced by transaction if using transaction DB.